src/handlers/rotate.rs

```diff
@@ -0,0 +1,142 @@
+use crate::util::image::{move_image, remove_cache_entries};
+use crate::{
+    util::{
+        auth::check_api_key,
+        image::{determine_img_dir, determine_img_path, save_image, ImageSearchBehaviour},
+    },
+    ServerState,
+};
+use axum::extract::{Query, State};
+use axum::headers::authorization::Bearer;
+use axum::headers::Authorization;
+use axum::http::StatusCode;
+use axum::TypedHeader;
+use libvips::{ops, VipsImage};
+use serde::Deserialize;
+use std::fs::rename;
+use uuid::Uuid;
+
+#[derive(Deserialize)]
+pub struct RotateQuery {
+    id: Uuid,
+    angle: f64,
+}
+
+fn compare_floats(a: f64, b: f64) -> bool {
+    (a - b).abs() < f64::EPSILON
+}
+
+fn is_valid_angle(angle: f64) -> bool {
+    compare_floats(angle, 0.0)
```

Comment:
Why is 0.0 an option? Imo we should only need 90, 180, 270 (aka -90). Could also be int instead of float

Reply:
hast recht